### ISSUE TITLE
[Bug]: NoMethodError: undefined method `[]' for nil:NilClass

### ISSUE THREAD

**Issue by alexantom** [Shopify/shopify-cli#2205]
### Please confirm that you have:

- [X] Searched [existing issues](https://github.com/Shopify/shopify-cli/issues) to see if your issue is a duplicate. (If you’ve found a duplicate issue, feel free to add additional information in a comment on it.)
- [X] Reproduced the issue in [the latest CLI version](https://github.com/Shopify/shopify-cli/releases).

### What type of project are you working on?

theme

### Expected behavior

`shopify theme serve` should have served the theme code

### Actual behavior

`shopify theme serve`. errored out

### Stack trace

```shell
/usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/lib/shopify_cli/theme/file.rb:115:in `visit_document': undefined method `[]' for nil:NilClass (NoMethodError)
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/lib/shopify_cli/theme/file.rb:104:in `normalize_json'
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/lib/shopify_cli/theme/file.rb:72:in `checksum'
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/lib/shopify_cli/theme/syncer/checksums.rb:20:in `file_has_changed?'
```
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/lib/shopify_cli/theme/syncer/json_update_handler.rb:19:in `block in enqueue_json_updates'
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/lib/shopify_cli/theme/syncer/json_update_handler.rb:19:in `select'
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/lib/shopify_cli/theme/syncer/json_update_handler.rb:19:in `enqueue_json_updates'
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/lib/shopify_cli/theme/syncer.rb:173:in `upload_theme!'
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/lib/project_types/theme/ui/sync_progress_bar.rb:11:in `public_send'
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/lib/project_types/theme/ui/sync_progress_bar.rb:11:in `block in progress'
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/vendor/deps/cli-ui/lib/cli/ui/progress.rb:34:in `progress'
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/lib/project_types/theme/ui/sync_progress_bar.rb:10:in `progress'
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/lib/project_types/theme/commands/serve.rb:27:in `block in call'
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/lib/shopify_cli/theme/dev_server.rb:55:in `block in start'
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/vendor/deps/cli-ui/lib/cli/ui/frame.rb:103:in `open'
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/lib/shopify_cli/theme/dev_server.rb:51:in `start'
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/lib/project_types/theme/commands/serve.rb:26:in `call'
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/lib/shopify_cli/command/sub_command.rb:16:in `call'
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/lib/shopify_cli/command.rb:27:in `call'
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/lib/shopify_cli/core/executor.rb:17:in `block (2 levels) in call'
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/vendor/deps/cli-kit/lib/cli/kit/executor.rb:37:in `block (2 levels) in with_logging'
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/vendor/deps/cli-ui/lib/cli/ui/stdout_router.rb:169:in `with_id'
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/vendor/deps/cli-kit/lib/cli/kit/executor.rb:36:in `block in with_logging'
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/vendor/deps/cli-ui/lib/cli/ui.rb:176:in `log_output_to'
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/vendor/deps/cli-kit/lib/cli/kit/executor.rb:35:in `with_logging'
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/lib/shopify_cli/core/executor.rb:16:in `block in call'
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/vendor/deps/cli-kit/lib/cli/kit/executor.rb:45:in `block (2 levels) in with_traps'
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/vendor/deps/cli-kit/lib/cli/kit/executor.rb:55:in `twrap'
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/vendor/deps/cli-kit/lib/cli/kit/executor.rb:44:in `block in with_traps'
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/vendor/deps/cli-kit/lib/cli/kit/executor.rb:55:in `twrap'
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/vendor/deps/cli-kit/lib/cli/kit/executor.rb:43:in `with_traps'
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/lib/shopify_cli/core/executor.rb:15:in `call'
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/lib/shopify_cli/core/entry_point.rb:24:in `block in call'
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/lib/shopify_cli/core/monorail.rb:26:in `log'
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/lib/shopify_cli/core/entry_point.rb:23:in `call'
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/bin/shopify:50:in `block (2 levels) in <top (required)>'
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/vendor/deps/cli-kit/lib/cli/kit/error_handler.rb:75:in `handle_abort'
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/vendor/deps/cli-kit/lib/cli/kit/error_handler.rb:21:in `call'
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/bin/shopify:49:in `block in <top (required)>'
	from /usr/local/Cellar/shopify-cli/2.15.2/gems/shopify-cli-2.15.2/bin/shopify:59:in `<top (required)>'
	from /usr/local/bin/shopify:7:in `load'
	from /usr/local/bin/shopify:7:in `<main>'
```


### Reproduction steps

1.  shopify theme serve

### Operating System

x86_64-darwin18

### CLI version (run `shopify version` if you're not sure)

2.15.2

### Shell

/bin/bash

### Ruby version (run `ruby -v` if you're not sure)

3.1.1p18

### What language and version are you using in your application?

_No response_

**Comment by corazzi** [comment]
@alexantom I had this issue and I figured out that it was caused by one of my JSON templates. 

Somehow, I'd managed to set the contents of a JSON template to just `null` (as in the word "null").

The line that throws the exception is here:

```
 114   │         def visit_document(value)
 115   │           visit_hash(value["sections"])
 116   │         end
```

Without digging further it looks like it's trying to read some section values or schema, which I wouldn't have had with an invalid JSON template. Check these 😄
